core/domain/src/main/java/com/google/samples/apps/nowinandroid/core/domain/GetSearchContentsUseCase.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.core.domain
+
+import com.google.samples.apps.nowinandroid.core.data.repository.SearchContentsRepository
+import com.google.samples.apps.nowinandroid.core.data.repository.SearchResult
+import com.google.samples.apps.nowinandroid.core.data.repository.UserDataRepository
+import com.google.samples.apps.nowinandroid.core.domain.model.FollowableTopic
+import com.google.samples.apps.nowinandroid.core.domain.model.UserNewsResource
+import com.google.samples.apps.nowinandroid.core.model.data.UserData
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import javax.inject.Inject
+
+/**
+ * A use case which returns the searched contents matched with the search query.
+ */
+class GetSearchContentsUseCase @Inject constructor(
+    private val searchContentsRepository: SearchContentsRepository,
+    private val userDataRepository: UserDataRepository,
+) {
+
+    operator fun invoke(
+        searchQuery: String,
+    ): Flow<UserSearchResult> =
+        searchContentsRepository.searchContents(searchQuery)
+            .mapToUserSearchResult(userDataRepository.userData)
+}
+
+fun Flow<SearchResult>.mapToUserSearchResult(userDataStream: Flow<UserData>): Flow<UserSearchResult> =
+    combine(userDataStream) { searchResult, userData ->
+        UserSearchResult(
+            topics = searchResult.topics.map { topic ->
+                FollowableTopic(
+                    topic = topic,
+                    isFollowed = topic.id in userData.followedTopics,
+                )
+            },
+            newsResources = searchResult.newsResources.map { news ->
+                UserNewsResource(
+                    newsResource = news,
+                    userData = userData,
+                )
+            },
+        )
+    }
+
+data class UserSearchResult(
```

Comment:
NIT: IMO it's beneficial to the code's readability to put classes in separate files

Reply:
Agreed. Done.